internal/worker/sshserver/listener_test.go

```diff
@@ -4,44 +4,139 @@
 package sshserver_test
 
 import (
+	"context"
+	"net"
+	"sync"
 	"time"
 
 	"github.com/juju/testing"
-	"google.golang.org/grpc/test/bufconn"
+	jc "github.com/juju/testing/checkers"
+	gomock "go.uber.org/mock/gomock"
 	gc "gopkg.in/check.v1"
-
-	"github.com/juju/juju/internal/worker/sshserver"
 )
 
+// sshServerListener is required to prevent a race condition
+// that can occur in tests.
+//
+// The SSH server tracks the listeners in use
+// but if the server's close() method executes
+// before we reach a safe point in the Serve() method
+// then the server's map of listeners will be empty.
+// A safe point to indicate the server is ready is
+// right before we start accepting connections.
+// Accept() will return with error if the underlying
+// listener is already closed.
+//
+// As such, we ensure accept has been called at least once
+// before allowing a close to take effect. The corresponding
+// piece to this is to receive from the closeAllowed channel
+// within your cleanup routine.
+//
+// Example:
```

Comment:
Let's remove this example, it isn't needed now and doc needs updating to remove the closeAllowed channel explanation.